core/include/scipp/core/dataset.h

```diff
@@ -913,11 +918,33 @@ class SCIPP_CORE_EXPORT DataArray {
     setCoord(dim, Variable(coord));
   }
 
-  template <class... Ts> auto slice(const Ts... slices) const {
-    return get().slice(slices...);
+  DataConstProxy slice(const Slice slice1) const & {
```

Comment:
Why did this need to change from the variadic approach used previously?

Reply:
It did not work with aggregate init, i.e., you had to use `Slice{...}` instead of, e.g., `{Dim::X, 4}`. This was inconsistent.

Reply:
That would require an extra set of `{`, but more importantly, a memory allocation.